packages/web-app-files/src/components/SideBar/Shares/Collaborators/NewCollaborator.vue

```diff
@@ -178,26 +173,50 @@ export default {
             }
 
             this.announcement = this.$_announcementWhenCollaboratorAdded
+
             return true
           })
         })
         .catch(error => {
-          console.log(error)
-          this.autocompleteInProgress = false
+          console.error(error)
+
+          this.searchInProgress = false
         })
+        .finally(() => {
+          this.searchInProgress = false
+        })
+    },
+
+    onSearch(query) {
+      this.autocompleteResults = []
+      this.searchQuery = query
+
+      if (query.length < this.minSearchLength) {
+        this.searchInProgress = false
+
+        return
+      }
+
+      this.searchInProgress = true
+
+      this.fetchRecipients(query)
     },
-    filterRecipients(item, queryText) {
-      if (item.value.shareType === shareTypes.remote) {
-        // done on server side
-        return true
+
+    filterRecipients(recipients, query) {
+      if (recipients.length < 1) {
+        return []
       }
-      return (
-        item.label.toLocaleLowerCase().indexOf(queryText.toLocaleLowerCase()) > -1 ||
-        item.value.shareWith.toLocaleLowerCase().indexOf(queryText.toLocaleLowerCase()) > -1 ||
-        (item.value.shareWithAdditionalInfo || '')
-          .toLocaleLowerCase()
-          .indexOf(queryText.toLocaleLowerCase()) > -1
-      )
+
+      return recipients.filter(recipient => {
```

Comment:
the brackets and return is not needed, `recipients.filter(recipient => recipient.value.shareType === shareTypes.remote ||...`